ui/tests/acceptance/custom-messages-auth-test.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'ember-qunit';
+import { click, visit } from '@ember/test-helpers';
+import { PAGE } from 'vault/tests/helpers/config-ui/message-selectors';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+
+const unauthenticatedMessageResponse = {
+  request_id: '664fbad0-fcd8-9023-4c5b-81a7962e9f4b',
+  lease_id: '',
+  renewable: false,
+  lease_duration: 0,
+  data: {
+    key_info: {
+      '02180e3f-bd5b-a851-bcc9-6f7983806df0': {
+        authenticated: false,
+        end_time: null,
+        link: {
+          title: '',
+        },
+        message: 'aGVsbG8gd29ybGQgaGVsbG8gd29scmQ=',
+        options: null,
+        start_time: '2024-01-04T08:00:00Z',
+        title: 'Banner title',
+        type: 'banner',
+      },
+      'a7d7d9b1-a1ca-800c-17c5-0783be88e29c': {
+        authenticated: false,
+        end_time: null,
+        link: {
+          title: '',
+        },
+        message: 'aGVyZSBpcyBhIGNvb2wgbWVzc2FnZQ==',
+        options: null,
+        start_time: '2024-01-01T08:00:00Z',
+        title: 'Modal title',
+        type: 'modal',
+      },
+    },
+    keys: ['02180e3f-bd5b-a851-bcc9-6f7983806df0', 'a7d7d9b1-a1ca-800c-17c5-0783be88e29c'],
+  },
+  wrap_info: null,
+  warnings: null,
+  auth: null,
+  mount_type: '',
+};
+
+module('Acceptance | auth custom messages auth tests', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+
+  test('it shows the alert banner and modal message', async function (assert) {
+    this.server.get('/sys/internal/ui/mounts', () => ({}));
+    this.server.get('/sys/internal/ui/unauthenticated-messages', function () {
+      return unauthenticatedMessageResponse;
+    });
+    await visit('/vault/auth');
+    const modalId = 'a7d7d9b1-a1ca-800c-17c5-0783be88e29c';
```

Comment:
suggestion to make this more legible: since these strings are random, we can set them to human-discernable strings so it's easier to match the ID's to the values in the mocked payload in the future